File: PrototypeIosPebbleOS/PrototypeIosObjC/PrototypeIosObjC/NotificationsViewController.m
//
//  NotificationsViewController.m
//  PrototypeIosObjC
//
//  Created by Kristina Gancheva on 27/11/14.
//  Copyright (c) 2014 Peperzaken. All rights reserved.
//
#import <UIKit/UIKit.h>
#import <QuartzCore/QuartzCore.h>
#import "NotificationsViewController.h"
#import "UIColorExtension.h"
#import "PebbleController.h"

@interface NotificationsViewController ()
@end

@implementation NotificationsViewController

#pragma mark - UIViewController method(s)

- (void)viewDidLoad {
    self.view.backgroundColor = [UIColor colorFromHex:@"#242d3cff"];
    
    self.textField.layer.cornerRadius = 10.0f;
    self.textField.backgroundColor = [UIColor colorFromHex: @"#7b818a"];
    self.textField.textColor = [UIColor whiteColor];

    [super viewDidLoad];
}

#pragma mark - IB actions

- (IBAction)notifyWatch:(id)sender {
    // send the user input to Pebble, notify the app user for the result by showing a popup
    [self.view endEditing:YES];
    
    NSString *popupMessage = @"\n\nPebble is not connected and notification couldn't be sent!\n\n";
    if ([[PebbleController sharedInstance] updateWatch:self.textField.text]) {
        popupMessage = @"\n\nNotification was sent.\n\n";
    }
    [self showPopup:popupMessage];
}

#pragma mark - Method for showing popup view

- (void)showPopup:(NSString *)popupMessage {
    UIAlertView *popup = [[UIAlertView alloc] initWithTitle:nil
                                                    message:popupMessage
                                                   delegate:nil
                                          cancelButtonTitle:nil
                                          otherButtonTitles:nil];
    [popup show];
    dispatch_time_t popTime = dispatch_time(DISPATCH_TIME_NOW, (int64_t)(1.0 * NSEC_PER_SEC)); // 1.0 indicates the delay in seconds
    dispatch_after(popTime, dispatch_get_main_queue(), ^(void){
        [popup dismissWithClickedButtonIndex:0 animated:YES];
        // after dismissing the popup, show previous view
        [self.presentingViewController dismissViewControllerAnimated:YES completion:nil];
    });
}

@end

File: PrototypeIosPebbleOS/PrototypeIosSwift/PrototypeIosSwift/NotificationsViewController.swift
//
//  NotificationsViewController.swift
//  PrototypeIosSwift
//
//  Created by Kristina Gancheva on 18/12/14.
//  Copyright (c) 2014 Peperzaken. All rights reserved.
//

import UIKit

class NotificationsViewController: UIViewController {

    @IBOutlet weak var textField: UITextField!
    @IBOutlet weak var dismissButton: UIButton!
    @IBOutlet weak var sendButton: UIButton!
    
    //MARK: UIViewController method(s)
    
    override func viewDidLoad() {
        super.viewDidLoad()
        self.view.backgroundColor = UIColor(rgba: "#242d3cff")
        textField.layer.cornerRadius = 10.0
        textField.backgroundColor = UIColor(rgba: "#7b818a")
        textField.textColor = UIColor.whiteColor()
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
    }
    
    //MARK: IB actions
    
    @IBAction func notifyWatch(sender: AnyObject) {
        // send the user input to Pebble, notify the app user for the result by showing a popup
        self.view.endEditing(true)
        
        var popupMessage = "\n\nPebble is not connected and notification couldn't be sent!\n\n"
        if PebbleController.sharedInstance.updateWatch(self.textField.text) {
            popupMessage = "\n\nNotification was sent.\n\n"
        }
        
        self.showPopup(popupMessage)
    }
    
    //MARK: Method for showing popup view
    private func showPopup(popupMessage: String) {
        var popup = UIAlertController(title: nil, message: popupMessage, preferredStyle: UIAlertControllerStyle.Alert)
        self.presentViewController(popup, animated: true, completion: nil)
        
        var popTime = dispatch_time(DISPATCH_TIME_NOW, Int64(1.0 * Double(NSEC_PER_SEC))) // 1.0 indicates the delay in seconds
        dispatch_after(popTime, dispatch_get_main_queue()) { () -> Void in
            self.dismissViewControllerAnimated(true, completion: nil)
            // after dismissing the popup, show previous view
            self.presentedViewController?.dismissViewControllerAnimated(true, completion: nil)
        }
    }
    
}